### ISSUE TITLE
DOCKER_ONLY for sysconfig

### ISSUE THREAD

**Issue by sgohl** [google/cadvisor#2895]
I decided to run cadviosor directly on the host, but I don't want to modify the systemd unit.
Instead, the `DOCKER_ONLY ` option should be made configurable via `/etc/sysconfig/cadvisor`, like

```
CADVISOR_DOCKER_ONLY="true"
```

**Comment by Creatone** [comment]
You mean ```--docker_only``` from https://github.com/google/cadvisor/blob/master/docs/runtime_options.md#limiting-which-containers-are-monitored?

**Comment by sgohl** [comment]
yes, exactly. The question is: What do I have to write into `/etc/sysconfig/cadvisor` to make it happen?

usually there is a flexible `$OPTIONS` variable which is appended in the systemd unit.

```
# cat /usr/lib/systemd/system/cadvisor.service
[Unit]
Description=cAdvisor

[Service]
EnvironmentFile=/etc/sysconfig/cadvisor
ExecStart=/usr/bin/cadvisor \
	--docker=${CADVISOR_DOCKER_ENDPOINT} \
	--port=${CADVISOR_PORT} \
	--storage_driver=${CADVISOR_STORAGE_DRIVER} \
	--storage_driver_host=${CADVISOR_STORAGE_DRIVER_HOST} \
	--storage_driver_password=${CADVISOR_STORAGE_DRIVER_PASSWORD} \
	--storage_driver_secure=${CADVISOR_STORAGE_DRIVER_SECURE} \
	--storage_driver_user=${CADVISOR_STORAGE_DRIVER_USER} \
	--logtostderr=${CADVISOR_LOG_TO_STDERR} \
        ${OPTIONS}
```

# cat /etc/sysconfig/cadvisor 
# config options for cAdvisor
#
#
OPTIONS="--docker_only"
```

see?

**Comment by Creatone** [comment]
It's working for you? For me it's fine.

**Comment by sgohl** [comment]
That was an example how it could be, but the systemd unit does not provide an `${CADVISOR_OPTIONS}` option.
the premise is that the systemd unit must not be modified manually

**Comment by Creatone** [comment]
We do not provide the systemd unit file. How you install cAdvisor?

```
the premise is that the systemd unit must not be modified manually
```

Why not?

**Comment by sgohl** [comment]
because of the unnecessary complications the modification of parts of an existing file in contrast to the simple and usual configuration method via sysconfig - this is why sysconfig actually exists.

I would either need to modify the systemd-unit via sed, or place the complete file while I only wanted to add a simple configuration flag. Both is ridiculous.

Think of provisioning like cloud-config, ignition etc
